stripe.go

```diff
@@ -1359,22 +1360,28 @@ func initUserAgent() {
 	if appInfo != nil {
 		encodedUserAgent += " " + appInfo.formatUserAgent()
 	}
+	encodedStripeUserAgentReady = &sync.Once{}
+}
 
-	stripeUserAgent := &stripeClientUserAgent{
-		Application:     appInfo,
-		BindingsVersion: clientversion,
-		Language:        "go",
-		LanguageVersion: runtime.Version(),
-		Publisher:       "stripe",
-		Uname:           getUname(),
-	}
-	marshaled, err := json.Marshal(stripeUserAgent)
-	// Encoding this struct should never be a problem, so we're okay to panic
-	// in case it is for some reason.
-	if err != nil {
-		panic(err)
-	}
-	encodedStripeUserAgent = string(marshaled)
+func getEncodedStripeUserAgent() string {
+	encodedStripeUserAgentReady.Do(func() {
```

Comment:
This way of caching would make `SetAppInfo` not update the stripeUserAgent multiple times. Although, I guess it can be fine when the assumption is that `SetAppInfo` is always called before using the client.

Reply:
We reset the `encodedStripeUserAgentReady` in `initUserAgent` which is called from `SetAppInfo`. So calling `SetAppInfo` will force the StripeUserAgent calculation to rerun.

Reply:
ah, yes, then it's fine